pkg/apis/pipeline/v1/pipeline_types.go

```diff
@@ -267,6 +267,29 @@ func (pt *PipelineTask) IsMatrixed() bool {
 	return pt.Matrix != nil && (pt.Matrix.hasParams() || pt.Matrix.hasInclude())
 }
 
+// extractAllParams extracts all the parameter in a Pipeline Task including Matrix parameter
+// and Matrix Include parameter
+func (pt *PipelineTask) extractAllParams() Params {
+	allParams := pt.Params
+	if pt.Matrix.hasParams() {
+		allParams = append(allParams, pt.Matrix.Params...)
+	}
+	if pt.Matrix.hasInclude() {
+		for _, include := range pt.Matrix.Include {
+			allParams = append(allParams, include.Params...)
+		}
+	}
+	return allParams
+}
+
+// extractAllParamValues extracts all the parameter values in a Pipeline Task including Matrix parameter
+// and Matrix Include parameter
+func (pt *PipelineTask) extractAllParamValues() []string {
+	allParams := pt.extractAllParams()
+	paramValues := allParams.extractParamValuesFromParams()
+	return paramValues
+}
+
```

Comment:
we don't need this function because it's just wrapping two functions that we can call directly
```suggestion
```